### PR TITLE
Feature/16295 convert breadcrumbs and tab item tests to rtl

### PR DESCRIPTION
**Pull request by jenniemc**
## Testing done
unit test

## Screenshots
n/a

## Acceptance criteria
- [ ] change unit test to use react testing library

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/TabItem.unit.spec.jsx [hunk 1]
```diff
@@ -17,7 +17,6 @@ describe('<TabItem>', () => {
     expect(tree.find('NavLink').props().to).to.equal('upcoming');
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/TabItem.unit.spec.jsx:17]: I think you can go ahead and delete this file

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx [hunk 2]
```diff
@@ -597,6 +597,242 @@ describe('VAOS integration: appointment list', () => {
...
+      screen.getByRole('tab', {
+        name: 'upcoming appointments',
+        selected: false,
+      }),
+    ).to.have.attribute('tabindex', '-1');
+
+    // triggers arrow right key
+
+    fireEvent.keyDown(
+      await screen.getByRole('tab', {
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx:660]: If you're using await here, you'd want to use `screen.findByRole`.

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx [hunk 3]
```diff
@@ -597,6 +597,242 @@ describe('VAOS integration: appointment list', () => {
...
+      screen.getByRole('tab', {
+        name: 'express care appointments',
+        selected: false,
+      }),
+    ).to.have.attribute('tabindex', '-1');
+
+    // triggers arrow left key
+
+    fireEvent.keyDown(
+      await screen.getByRole('tab', {
```
Comment (jbalboni) [inline src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx:756]: Same as above, with await we'd use `screen.findByRole`

**Review by jbalboni — COMMENTED** [review]
Couple minor comments, looks good otherwise

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/TabItem.unit.spec.jsx (hunk 1 above)
Comment (jenniemc) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/TabItem.unit.spec.jsx]: Done

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx (hunk 2 above)
Comment (jenniemc) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx]: Done

File: src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx (hunk 3 above)
Comment (jenniemc) [reply src/applications/vaos/tests/appointment-list/components/AppointmentsPage/index.unit.spec.jsx]: Done

**Review by jbalboni — APPROVED** [review]

**Pull request merged**